GalaxyZooTouchTable/GalaxyZooTouchTable/ViewModels/ClassificationPanelViewModel.cs

```diff
@@ -119,27 +116,50 @@ public AnswerButton SelectedAnswer
             }
         }
 
-        public ClassificationPanelViewModel(Workflow workflow, TableUser user, ObservableCollection<TableUser> allUsers)
+        public ClassificationPanelViewModel(IPanoptesRepository panoptesRepo, IGraphQLRepository graphQLRepo, TableUser user)
         {
-            if (workflow != null)
-            {
-                PrepareForNewClassification();
-            }
-            LoadCommands();
-            Workflow = workflow;
+            _panoptesRepository = panoptesRepo;
+            _graphQLRepository = graphQLRepo;
             User = user;
-            CurrentTask = workflow.Tasks[workflow.FirstTask];
-            CurrentTaskIndex = workflow.FirstTask;
-            Notifications = new NotificationsViewModel(user, allUsers, this);
+
+            Notifications = new NotificationsViewModel(user);
             LevelerViewModel = new LevelerViewModel(user);
-            StillThere = new StillThereViewModel(this);
+            StillThere = new StillThereViewModel();
+
+            AddSubscribers();
+            GetWorkflow();
+            LoadCommands();
+        }
+
+        private void AddSubscribers()
+        {
             ExamplesViewModel.PropertyChanged += ResetTimer;
             LevelerViewModel.PropertyChanged += ResetTimer;
+            Notifications.GetSubjectById += OnGetSubjectById;
+            Notifications.ChangeView += OnChangeView;
+            Notifications.SendRequestToUser += OnSendRequestToUser;
+            StillThere.ResetFiveMinuteTimer += ResetTimer;
+            StillThere.CloseClassificationPanel += CloseClassificationPanel;
+        }
 
-            if (CurrentTask.Answers != null)
-            {
-                CurrentAnswers = ParseTaskAnswers(CurrentTask.Answers);
-            }
+        private void CloseClassificationPanel()
```

Comment:
Seems like you can just get rid of this function if all it does is just call another function

Reply:
Oh duh. This was expecting an `object sender` to be passed through the delegate.